### PR TITLE
Install and use toxiproxy in vagrant

### PR DESCRIPTION
**Pull request by eapache**
This sends the public zk ports (2181-2185) and kafka ports (6667-6671) through
toxiproxy.

### CODE REVIEW

**Comment by eapache** [comment]
Total tangent, but ever since Willem disabled `auto.create.topics`, it's been clear that the heisenbug is *not* in sarama proper and is just the topics not being created in the cluster for some reason...

**Comment by wvanbergen** [comment]
Yeah, I think the topics failed to create for whatever reason, and then
would get created with the default number of partitions

**Comment by eapache** [comment]
@Shopify/kafka this doesn't add any tests with toxiproxy, but it does set it up in vagrant and CI for future use. We may have to bug @Sirupsen to bless the toxiproxy go API, as it is still labelled "WIP".

**Comment by eapache** [comment]
(This is ready for review now)

File: vagrant/run_toxiproxy.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+wget -S --post-data='{"name":"zk2", "upstream":"localhost:21802", "listen":"0.0.0.0:2182"}' localhost:8474/proxies
+wget -S --post-data='{"name":"zk3", "upstream":"localhost:21803", "listen":"0.0.0.0:2183"}' localhost:8474/proxies
+wget -S --post-data='{"name":"zk4", "upstream":"localhost:21804", "listen":"0.0.0.0:2184"}' localhost:8474/proxies
+wget -S --post-data='{"name":"zk5", "upstream":"localhost:21805", "listen":"0.0.0.0:2185"}' localhost:8474/proxies
+
+wget -S --post-data='{"name":"kafka1", "upstream":"localhost:26667", "listen":"0.0.0.0:6667"}' localhost:8474/proxies
+wget -S --post-data='{"name":"kafka2", "upstream":"localhost:26668", "listen":"0.0.0.0:6668"}' localhost:8474/proxies
+wget -S --post-data='{"name":"kafka3", "upstream":"localhost:26669", "listen":"0.0.0.0:6669"}' localhost:8474/proxies
+wget -S --post-data='{"name":"kafka4", "upstream":"localhost:26670", "listen":"0.0.0.0:6670"}' localhost:8474/proxies
+wget -S --post-data='{"name":"kafka5", "upstream":"localhost:26671", "listen":"0.0.0.0:6671"}' localhost:8474/proxies
```
Comment (Sirupsen) [inline vagrant/run_toxiproxy.sh]: yeah this sucks... eventually the Go wrapper will have a helper.

**Comment by Sirupsen** [comment]
LGTM

The Go API might change a bit, and needs some more implemented, but overall it should be fine to use now

**Comment by wvanbergen** [comment]
While you are working on this, can we run Kafka on port 9091...9095 ?

**Comment by wvanbergen** [comment]
Other than that this looks good to me. :shipit:

**Comment by eapache** [comment]
Ports moved, and a bit more boot time granted, time for any final comments.

**Comment by eapache** [comment]
OK, I had to push some more tweaks to get everything working in both vagrant and CI, another look please.

**Comment by wvanbergen** [comment]
:+1:

**Pull request merged**